app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -874,7 +874,14 @@ export default Ember.Controller.extend({
       if (key === "new_topic") {
         this.send("clearTopicDraft");
       }
-      Draft.clear(key, this.get("model.draftSequence"));
+
+      Draft.clear(key, this.get("model.draftSequence")).then(() => {
+        if (
+          this.get("application.currentRouteName") === "userActivity.drafts"
```

Comment:
@pmusaraj I do believe that checking the route name is the wrong approach there. My main issue with it is it is putting code that belongs in another route in this controller.

What I would recommend is to use an appEvent. You can trigger something like `this.appEvents.trigger('draft:destroyed', key)`

Then in the userActivity.drafts route, listen for the app event and call refresh.